### PR TITLE
HDDS-11637. Compile failure is ignored in build check

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Exit code in `build.sh` (used by `build` check) was accidentally omitted in HDDS-11057.  This results in "successful" build check even in case of build failure.  (`dependency` or `acceptance` checks may still catch the problem due to missing artifacts.)

This PR makes `build.sh` exit with failure result if there are any errors.

Also fix a javadoc [failure](https://github.com/apache/ozone/actions/runs/11549582108/job/32143421325#step:8:15382), which was not caught by CI earlier due to the problem above.

https://issues.apache.org/jira/browse/HDDS-11637

## How was this patch tested?

Ran the script after introducing some syntax error:

```
$ echo "garbage" >> hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/HddsConfigKeys.java
$ ./hadoop-ozone/dev-support/checks/build.sh; echo $?
...
[ERROR] hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/HddsConfigKeys.java:[405,1] reached end of file while parsing
...
[INFO] BUILD FAILURE
...
1
```

### CODE REVIEW

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/upgrade/UpgradeActionRecon.java [hunk 1]
```diff
@@ -42,15 +42,12 @@
...
- *     @Override
- *     public void execute() throws Exception {
- *       // Custom upgrade logic for FEATURE_1
- *     }
- *  }
+ * &#64;UpgradeActionRecon(feature = FEATURE_NAME, type = FINALIZE)
+ * public class FeatureNameUpgradeAction implements ReconUpgradeAction {
+ *   &#64;Override
+ *   public void execute() throws Exception {
+ *     // Custom upgrade logic for FEATURE_1
```
Comment (ArafatKhan2198) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/upgrade/UpgradeActionRecon.java:49]: Thanks a lot, @adoroszlai, for fixing this JavaDoc!

I have one small request: could you please rename "FEATURE_1" to "FEATURE_NAME"? I missed it last time. Thank you!

**Review by ArafatKhan2198 — APPROVED** [review]
LGTM

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/upgrade/UpgradeActionRecon.java (hunk 1 above)
Comment (adoroszlai) [reply hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/upgrade/UpgradeActionRecon.java:49]: I really don't want to run full CI just to fix this existing typo.  Please make a note of the problem and fix it when you revisit this file for other reasons.

**Comment by adoroszlai** [comment]
Thanks @ArafatKhan2198 for the review.

**Pull request merged**